framework/assets/yii.activeForm.js

```diff
@@ -195,6 +198,18 @@
                     });
                     $form.on('submit.yiiActiveForm', methods.submitForm);
                 }
+                if (settings.scrollToError) {
+                    $form.on('afterValidate.yiiActiveForm', function(event, messages, errorAttributes) {
```

Comment:
I added errorAttributes so that custom scrolling code has access to the same data. From there it made more sense to rewrite the default scrolling logic the same way you'd write your own custom logic.